api-test/internal/config/config.go

```
@@ -0,0 +1,826 @@
+// legacyKeyRenames maps config keys this harness used to accept to their current
+// names. json.Unmarshal ignores unknown keys, so without this an overlay still
+// carrying the old name loses that whole block in silence — and api.tls_verify
+// defaults back to true, which fails closed against a self-signed deployment and
+// reads as a certificate problem rather than a stale config.
+var legacyKeyRenames = map[string]string{"bdd": "api"}
+
+// rejectLegacyKeys reports a config still using a pre-rename top-level key.
+func rejectLegacyKeys(path string, data []byte) error {
+	var top map[string]json.RawMessage
+	if err := json.Unmarshal(data, &top); err != nil {
+		return nil // malformed JSON is reported by the real decode
```

golangci-lint (2.12.2)

[error] error is not nil (line 250) but it returns nil

(nilerr)